src/cpp/session/modules/rmarkdown/NotebookChunkDefs.cpp

```diff
@@ -54,7 +57,16 @@ SEXP rs_getRmdWorkingDir(SEXP rmdFileSEXP, SEXP docIdSEXP)
    std::string workingDir;
    getChunkValue(docPath, docId, kChunkWorkingDir, &workingDir);
    if (!workingDir.empty())
+   {
       dir = module_context::resolveAliasedPath(workingDir);
+   }
+   else
+   {
+      // if this is a Quarto document, attempt to look up its desired
+      // working directory that way (returns an empty FilePath if not
+      // a Quarto doc or no explicit working directory is set)
+      dir = quarto::getQuartoExecutionDir(docPath);
```

Comment:
Do you think this call should be guarded by something like `isQuartoDocument()`? I'm mainly thinking of all the stuff happening here to detect "Quarto Markdown". https://github.com/rstudio/rstudio/blob/f2d353ccbdc7f9813a4e081809427e94eb22fc35/src/cpp/session/modules/quarto/SessionQuarto.cpp#L297-L347

Reply:
Ugh, that's a lot of heuristics, but the goal here is to make completion and inline execution fully symmetric with rendering, and as far as I can see the above is the logic we use when deciding whether to render with Quarto so we probably need it. I'll see if there's an easy way to hook it in w/o duplicating.

Reply:
Update: Addressed now.